packages/wrangler/src/module-collection.ts

```diff
@@ -101,6 +81,69 @@ export default function createModuleCollector(props: {
 	// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
 	rulesToRemove.forEach((rule) => rules!.splice(rules!.indexOf(rule), 1));
 
+	return { rules, rulesToRemove };
+}
+
+export async function matchFiles(
+	files: string[],
+	rules: Config["rules"],
+	rulesToRemove: Config["rules"]
+) {
+	const modules: CfModule[] = [];
+	for (const rule of rules) {
+		for (const glob of rule.globs) {
+			const regexp = globToRegExp(glob);
+			modules.push(
+				...(await Promise.all(
+					files
+						.filter((f) => regexp.test(f))
+						.map(async (name) => {
+							const filePath = name;
+							const fileContent = await readFile(filePath);
+
+							return {
+								name: filePath,
+								content: fileContent,
+								type: RuleTypeToModuleType[rule.type],
+							};
+						})
+				))
+			);
+		}
+	}
+	for (const rule of rulesToRemove) {
```

Comment:
A comment here explaining that we are now just sanity checking the files that would have matched rules that were removed.